packages/lib-classifier/src/store/AnnotatedSteps/AnnotatedSteps.js

```diff
@@ -0,0 +1,148 @@
+import cuid from 'cuid'
+import { getRoot, tryReference, types } from 'mobx-state-tree'
+import { UndoManager } from 'mst-middlewares'
+
+import AnnotatedStep from './AnnotatedStep'
+
+let undoManager = {}
+function setUndoManager(targetStore) {
+    undoManager = UndoManager.create({}, { targetStore })
+}
+/**
+  A history manager which records completed steps, from store.workflowSteps, and annotations, from store.classifications.
+*/
+const AnnotatedSteps = types.model('AnnotatedSteps', {
+  /** Completed steps, with their annotations */
+  steps: types.map(AnnotatedStep)
+})
```

Comment:
Shaun's dev notes 2:
- OK, I can mostly grok this now.
- Tasks/Task components now required Annotated Steps.
- Annotated Steps is a logical container that manages where we are in a series of Annotated Step(s)
- An Annotated Step is just a classic Step with the annotations. (To grok: so what happened to classification.annotation? It's still there, so does _AnnotatedStep.annotations_ just point to the same reference, or a sub-tree of classification.annotation ? Check/re-read.)
- So _Task_ has _AnnotatedSteps_ has many _AnnotatedStep(s)_ is _Step_ + annotations.

Reply:
If you take snapshots, `AnnotatedSteps.steps` looks something like this:
```
[
  { id: 'ckm2o6nnr000…', step: 'S2', annotations: ['ckm2o6nnr000h3h62y7f568c2', 'ckm2o6l7700053h624zzdbner'] }
]
```

It records identifiers, which point to models elsewhere in the store. MST automatically resolves those to the corresponding models in `rootStore.workflowSteps` and `rootStore.classifications`. `AnnotatedStep` here is being used to tie a step key to a list of annotation IDs. A given workflow step key might be annotated zero, once or many times so step keys aren't used as identifiers.